pkgs/development/interpreters/nextflow/default.nix

```diff
@@ -1,54 +1,102 @@
-{ lib
-, stdenv
-, fetchurl
-, makeWrapper
-, openjdk17
-, wget
-, which
-, gnused
-, gawk
-, coreutils
-, buildFHSEnv
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  openjdk,
+  gradle,
+  wget,
+  which,
+  gnused,
+  gawk,
+  coreutils,
+  testers,
+  nextflow,
+  bash,
 }:
-
-let
-  nextflow =
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: rec {
   pname = "nextflow";
-  version = "22.10.6";
+  version = "24.08.0-edge";
 
   src = fetchurl {
-    url = "https://github.com/nextflow-io/nextflow/releases/download/v${version}/nextflow-${version}-all";
-    hash = "sha256-zeYsKxWRnzr0W6CD+yjoAXwCN/AbN5P4HhH1oftnrjY=";
+    url = "https://github.com/nextflow-io/nextflow/archive/6e866ae81ff3bf8a9729e9dbaa9dd89afcb81a4b.tar.gz";
+    hash = "sha256-j1HFHn8Vu1/HLC1aAruFV7wm3otze8fTegAcf+cLrY4=";
   };
 
   nativeBuildInputs = [
     makeWrapper
-    openjdk17
+    gradle
+    bash
+    openjdk
     wget
     which
     gnused
     gawk
     coreutils
   ];
 
-  dontUnpack = true;
+  mitmCache = gradle.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+  __darwinAllowLocalNetworking = true;
+
+  patches = [ ./add-build-dependencies.patch ];
+
+  postPatch = ''
+    # Ironically, replacing 'bin/bash' by '/usr/bin/env [-S]' would work when
+    # actually using `nextflow` (in NixOS). However, that invocation fails
+    # within any Nix build environment (because there is no /usr/...). Thus,
+    # invoking `nextflow` within nixpkgs would be impossible.
+    substituteInPlace modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy \
```

Comment:
I don't think this will work. The groovy file is not [a script](https://github.com/NixOS/nixpkgs/blob/e1781b7c74f9f8e9606a6f9e422c4b51c6c49540/pkgs/stdenv/generic/setup.sh#L497) (and it's not about the firste line in the file but rather a string constant that is problematic) but `patchShebang` [wants](https://github.com/NixOS/nixpkgs/blob/19d4f7dc485f74109bd66ef74231285ff797a823/pkgs/build-support/setup-hooks/patch-shebangs.sh#L51) one.

Reply:
that's why i said install phase, not patch phase

Reply:
@chayleaf My, maybe we are misunderstanding each other. The line I have to patch is this [here](https://github.com/nextflow-io/nextflow/blob/6e866ae81ff3bf8a9729e9dbaa9dd89afcb81a4b/modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy#L87). This is not a shebang and it is neither used as one (i.e. not copied into a template file). From how I understand `patchShebang` (the bash function) works, it will not patch that file, no matter when/how I invoke it. It is not designed to replace arbitrary occurrences of strings that resemble a shebang within a given file but only the first line in "scripts" (see my links above).

Reply:
you can patch the resulting bash wrapper, not the groovy file

Reply:
IMHO, there is no resulting file that could be patched. `nextflow` generates the shell script on the fly (I believe this happens [here](https://github.com/nextflow-io/nextflow/blob/6e866ae81ff3bf8a9729e9dbaa9dd89afcb81a4b/modules/nextflow/src/main/groovy/nextflow/executor/local/LocalTaskHandler.groovy#L134)). Note that nextflow is a workflow engine, it generates a task graph and executes the tasks. A task is eventually only a (generated) script. This script is then executed. The string constant ("/bin/bash") is also used at other locations ([ex](https://github.com/nextflow-io/nextflow/blob/master/modules/nextflow/src/main/groovy/nextflow/processor/TaskBean.groovy#L136)).